### PR TITLE
[CI-only] Update build trigger to run on pushes to main and the release branches

### PR DESCRIPTION
**Pull request by mdeggies**
This updates nomad to always builds on commits to `main` and the release branches. This will avoid backport issues in the build workflow (from `main` into the release branches) and let us avoid having to manually allowlist new release branches in the build workflow. This also matches what is in the release configuration file: https://github.com/hashicorp/nomad/blob/main/.release/ci.hcl#L14-L17.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -2,8 +2,9 @@ name: build
 
 on:
   push:
-    branches:
-      - "main"
+    branches-ignore:
+      - docs/**
+      - backport/docs/**
```
Comment (tgross) [inline .github/workflows/build.yml:7]: We don't run builds tests for branches prefixed by E2E, as those changes only impact our nightly E2E suite (and takes 1hr+):

```suggestion
    branches-ignore:
      - docs/**
      - backport/docs/**
      - e2e-*
```

**Review by claire-labry — APPROVED** [review]

**Review by DerekStrickland — APPROVED** [review]
:shipit:

**Comment by tgross** [comment]
Leaving a note that there's still some open discussion on this one internally: https://hashicorp.slack.com/archives/CUYKT2A73/p1663623216766039

**Comment by mdeggies** [comment]
I updated this PR to always trigger builds in `main` and the `release/**` branches. This will avoid backport issues in the build workflow (from `main` into the release branches) and let us avoid having to manually allowlist new release branches in the build workflow. This also matches what is in the release configuration file: https://github.com/hashicorp/nomad/blob/main/.release/ci.hcl#L14-L17.

**Review by tgross — APPROVED** [review]
:+1: LGTM

**Pull request merged**